### PR TITLE
Add tests and fix rand(MersenneTwister, dims)

### PR DESCRIPTION
**Pull request by felixcremer**
This is to add some tests and I fixed the rand with an RNG but without the type information.
I will continue adding to this this morning, but I wanted to already run it on CI.

### CODE REVIEW

**Comment by felixcremer** [comment]
This is good to go, I worked on the other branches rather than these tests.

File: test/utils.jl [hunk 1]
```diff
@@ -97,6 +97,7 @@ end
     @test reo_s == s
     @test dims(reo_s) == dims(s)
 
+    @test_throws MethodError reorder(da, rand(2,3))
```
Comment (rafaqz) [inline test/utils.jl:100]:
```suggestion
```

Is this needed here? seems unrelated

**Review by rafaqz — APPROVED** [review]

File: test/utils.jl (hunk 1 above)
Comment (felixcremer) [reply test/utils.jl:100]: This is unrelated to the fixes of rand but I actual started this PR to work on test coverage therefore it ended up in there. I actually misunderstood how the dispatch in reorder works. Therefore yes we can just remove it.
